ui-admin/src/study/participants/enrolleeView/EnrolleeTimeline.tsx

```diff
@@ -0,0 +1,147 @@
+  const [isLoading, setIsLoading] = useState(true)
+
+  /** matches each notification to a corresponding config by id */
+  function attachConfigsToNotifications(rawNotifications: Notification[]) {
+    rawNotifications.forEach(notification => {
+      notification.notificationConfig = currentEnv.notificationConfigs
+        .find(config => config.id === notification.notificationConfigId)
+    })
+  }
+
+  const notificationToTableEntry = (val: Notification): EventNotificationTableEntry => {
+    return {
+      id: val.id,
+      deliveryType: val.deliveryType,
+      deliveryStatus: val.deliveryStatus,
+      notificationConfig: val.notificationConfig,
+      notificationConfigId: val.notificationConfigId,
+      createdAt: val.createdAt
+    }
+  }
+
+  const eventToTableEntry = (val: Event): EventNotificationTableEntry => {
+    return {
+      id: val.id,
+      createdAt: val.createdAt,
+      eventClass: val.eventClass
+    }
+  }
+
+  useEffect(() => {
+    Api.fetchEnrolleeNotifications(portal.shortcode, study.shortcode, currentEnv.environmentName, enrollee.shortcode)
+      .then(response => {
+        attachConfigsToNotifications(response)
+        setNotifications(response)
+        Api.fetchEnrolleeEvents(portal.shortcode, study.shortcode, currentEnv.environmentName, enrollee.shortcode)
+          .then(response => {
+            setEvents(response)
+            setIsLoading(false)
+          })
+      })
+  }, [enrollee.shortcode])
+
+  useEffect(() => {
+    const newTableData: EventNotificationTableEntry[] = []
+
+    newTableData.push(...notifications.map(notificationToTableEntry))
+    newTableData.push(...events.map(eventToTableEntry))
+
+    const sorted = newTableData.sort((obj1, obj2) => {
+      // sort function is "backwards" so it sorts descending
+      // (most recent event is on top)
+      if (obj1.createdAt < obj2.createdAt) {
+        return 1
+      }
+
+      if (obj1.createdAt > obj2.createdAt) {
+        return -1
+      }
+
+      return 0
+    })
+
+    setTableData(sorted)
+  }, [events, notifications])
+
+  return <div>
+    <h5>Timeline</h5>
+    <LoadingSpinner isLoading={isLoading}>
+      <table className="table table-striped">
+        <thead>
+          <tr>
+            <th>notification/event</th>
+            <th>method</th>
+            <th>status</th>
+            <th>time</th>
+            <th>config</th>
+          </tr>
+        </thead>
+        <tbody>
+          {tableData.map(row => {
+            const matchedConfig = currentEnv.notificationConfigs
+              .find(cfg => cfg.id === row.notificationConfigId)
+            return <tr key={row.id}>
+              <td>
+                {row.notificationConfig && <NotificationConfigTypeDisplay config={row.notificationConfig}/>}
+                {row.eventClass && eventClassToHumanReadable(row.eventClass)}
+              </td>
+
+              <td>
+                {row.deliveryType && row.deliveryType}
```

Comment:
IMO using the nullish coalesce operator is clearer at communicating "show this, but show blank if undefined"
```suggestion
                {row.deliveryType ?? ''}
```